src/config/SourcegraphConfigOptions.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+final class PhabricatorSourcegraphConfigOptions extends PhabricatorApplicationConfigOptions
+{
+    public function getName()
+    {
+        return pht('Sourcegraph');
+    }
+
+    public function getDescription()
+    {
+        return pht('Configure Sourcegraph Settings.');
+    }
+
+    public function getIcon()
+    {
+        return 'fa-asterisk';
+    }
+
+    public function getGroup()
+    {
+        return 'apps';
+    }
+
+    public function getOptions()
+    {
+        $repo_type = 'sourcegraph.repo';
+        $repo_example = array(
+            array(
+                'path' => 'gorilla/mux',
+                'callsign' => 'MUX',
+            ),
+        );
+
+        $repo_example = id(new PhutilJSON())->encodeAsList(
+            $repo_example);
+
+        return array(
+            $this->newOption(
+                'sourcegraph.url',
+                'string',
+                null)
+                ->setDescription(pht('URL to Sourcegraph.'))
+                ->addExample('https://sourcegraph.example.com', pht('Valid Setting')),
+            $this->newOption('sourcegraph.repos', $repo_type, array())
```

Comment:
The name `sourcegraph.repos` is pretty general and might be confusing. It would be better to be specific, like naming it `sourcegraph.callsignMapping`.